src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Flows/AcquireTokenHandlerBase.cs

```diff
@@ -110,7 +110,13 @@ protected AcquireTokenHandlerBase(RequestData requestData)
             brokerParameters[BrokerParameter.Resource] = requestData.Resource;
             brokerParameters[BrokerParameter.ClientId] = requestData.ClientKey.ClientId;
             brokerParameters[BrokerParameter.CorrelationId] = this.CallState.CorrelationId.ToString();
-            brokerParameters[BrokerParameter.ClientVersion] = AdalIdHelper.GetAssemblyFileVersion();
+
+            var adalVersion = AdalIdHelper.GetAdalVersion();
+            if (AdalIdHelper.VersionNotDetermined.Equals(adalVersion))
```

Comment:
What is the logic here? is it related to the hotfix?

Reply:
this is part of the fix that Petro did for 19.0

---
In reply to: [164965585](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/972#discussion_r164965585) [](ancestors = 164965585)